src/main/java/uk/gov/justice/hmpps/datacompliance/repository/jpa/model/referral/OffenderDeletionReferral.java

```diff
@@ -91,13 +91,13 @@ public OffenderDeletionReferral(final Long referralId,
     private LocalDateTime receivedDateTime;
 
     @OneToMany(mappedBy = "offenderDeletionReferral", cascade = PERSIST, fetch = LAZY)
-    private final List<ReferredOffenderBooking> offenderBookings = new ArrayList<>();
+    private final List<ReferredOffenderIds> offenderIds = new ArrayList<>();
 
     @OneToOne(mappedBy = "offenderDeletionReferral", cascade = PERSIST, fetch = LAZY)
     private ReferralResolution referralResolution;
 
-    public OffenderDeletionReferral addReferredOffenderBooking(final ReferredOffenderBooking booking) {
-        this.offenderBookings.add(booking);
+    public OffenderDeletionReferral addReferredOffenderIds(final ReferredOffenderIds booking) {
```

Comment:
This code is quite confusing.  Your passing in a singular referred offender IDs that you're calling a booking to then add to the list of offender IDs that isn't a list a offender IDs.  A backref is then set on the booking and the deletion referral is returned even though it is never used?

Would be better to do something like
```java
    public void addReferredOffenderId(final ReferredOffenderId referredOffenderId) {
        referredOffenderIds.add(referredOffenderId);
        referredOffenderId.setOffenderDeletionReferral(this);
    }
```
although I'm not happy about that either.  `ReferredOffenderId` sounds like it is the offender id of a referral but it isn't.  Maybe `OffenderReferral` instead?

Reply:
Agreed, it didn't help that I hadn't properly renamed everything, sorry!!  I'm now using `ReferredOffenderAlias`